From 2009f823a801afd0ffbe65960b0a002c71862253 Mon Sep 17 00:00:00 2001
From: Christian Hohnstaedt <chohnstaedt@innominate.com>
Date: Mon, 17 Nov 2014 16:15:42 +0100
Subject: [PATCH] RedHat Bug #1164340 - segfault when viewing a RHEL
 entitlement certificate

...in X509V3_EXT_get v3_lib.c:15

Fix the x509v3ext copy constructor

Closes: #770091
---
 lib/x509v3ext.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/lib/x509v3ext.cpp b/lib/x509v3ext.cpp
index d94cbda..87bfc63 100644
--- a/lib/x509v3ext.cpp
+++ b/lib/x509v3ext.cpp
@@ -26,10 +26,9 @@ x509v3ext::x509v3ext(const X509_EXTENSION *n)
 
 x509v3ext::x509v3ext(const x509v3ext &n)
 {
-	ext = NULL;
-	if (!n.isValid())
-		return;
-	set(n.ext);
+	ext = X509_EXTENSION_new();
+	if (n.ext && n.ext->value && n.ext->value->length > 0)
+		set(n.ext);
 }
 
 x509v3ext::~x509v3ext()
-- 
2.1.3

